src/main/scala/com/thenewmotion/akka/rabbitmq/ConnectionActor.scala

```diff
@@ -124,7 +124,7 @@ class ConnectionActor(
     log.debug("{} closing broken connection {}", self.path, connection)
     closeIfOpen(connection)
 
-    self ! Connect
+    setTimer(reconnectTimer, Connect, reconnectionDelay / 3, repeat = false)
```

Comment:
I post some logs in the issue report. Maybe you could find some useful info there.

The problem I'm facing it real. I can easily reproduce it. The close operation on the extra channel cause some unexpected exception.

After this fix. the problem gone.